marathon_acme/tests/test_fake_marathon.py

```diff
@@ -70,61 +62,61 @@ def test_get_apps(self):
         }
         self.marathon.add_app(app)
 
-        response = yield self.client.request('GET', '/v2/apps')
-        self.assertThat(response, IsJsonResponseWithCode(200))
+        response = self.client.get('http://localhost/v2/apps')
+        assert_that(response, succeeded(MatchesAll(
+            IsJsonResponseWithCode(200),
+            After(json_content, succeeded(Equals({'apps': [app]})))
+        )))
 
-        response_json = yield json_content(response)
-        self.assertThat(response_json, Equals({'apps': [app]}))
-
-    @inlineCallbacks
     def test_get_events(self):
         """
         When a request is made to the event stream endpoint, an SSE stream
         should be received in response and an event should be fired that
         indicates that the stream was attached to.
         """
-        response = yield self.client.request('GET', '/v2/events', headers={
+        response = self.client.get('http://localhost/v2/events', headers={
             'Accept': 'text/event-stream'
         })
-        self.assertThat(response, IsSseResponse())
-
-        data = []
-        sse_content(response, {'event_stream_attached': data.append})
-
-        yield wait0()
 
-        self.assertThat(len(data), Equals(1))
+        assert_that(response, succeeded(MatchesAll(
+            IsSseResponse(),
+            After(
+                partial(collect_events, 'event_stream_attached'),
+                MatchesListwise([
+                    After(json.loads, IsMarathonEvent(
+                        'event_stream_attached',
+                        clientIp=Equals('127.0.0.1')))
+                ])
+            )
+        )))
 
-        data_json = json.loads(data[0])
-        # FIXME: No clientIp in request
-        self.assertThat(data_json, IsMarathonEvent(
-            'event_stream_attached', remoteAddress=Is(None)))
-
-    @inlineCallbacks
     def test_get_events_lost_connection(self):
         """
         When two connections are made the the event stream, the first
         connection should receive events for both connections attaching to the
         stream. Then, when the first connection is disconnected, the second
         should receive a detach event for the first.
         """
-        response1 = yield self.client.request('GET', '/v2/events', headers={
+        response1_d = self.client.get('http://localhost/v2/events', headers={
             'Accept': 'text/event-stream'
         })
-        self.assertThat(response1, IsSseResponse())
+        # First listener attaches and receives event it attached
+        assert_that(response1_d, succeeded(IsSseResponse()))
 
+        response1 = response1_d.result
```

Comment:
If we're going to do this sort of thing in other tests, maybe a little helper function that lets us do `response1 = assert_sse_response(self.client.get(...))` might be useful?

If it's just the two instances in this one test, it's probably not worth the effort. Even if it is a thing we want to do, it probably belongs in a later "test cleanup" PR.